operations/mimir/autoscaling.libsonnet

```diff
@@ -141,17 +143,22 @@
     // Simple method to convert binary multiples.
     // Only works for limited set of SI prefixes (as below).
 
-    if std.endsWith(str, 'Ki') then (
-      std.parseJson(std.rstripChars(str, 'Ki')) * std.pow(2, 10)
-    ) else if std.endsWith(str, 'Mi') then (
-      std.parseJson(std.rstripChars(str, 'Mi')) * std.pow(2, 20)
-    ) else if std.endsWith(str, 'Gi') then (
-      std.parseJson(std.rstripChars(str, 'Gi')) * std.pow(2, 30)
-    ) else if std.endsWith(str, 'Ti') then (
-      std.parseJson(std.rstripChars(str, 'Ti')) * std.pow(2, 40)
-    ) else (
-      str
-    )
+    // Utility converting the input to a (potentially decimal) number of bytes
+    local siToBytesDecimal(str) = (
+      if std.endsWith(str, 'Ki') then (
+        std.parseJson(std.rstripChars(str, 'Ki')) * std.pow(2, 10)
+      ) else if std.endsWith(str, 'Mi') then (
+        std.parseJson(std.rstripChars(str, 'Mi')) * std.pow(2, 20)
+      ) else if std.endsWith(str, 'Gi') then (
+        std.parseJson(std.rstripChars(str, 'Gi')) * std.pow(2, 30)
+      ) else if std.endsWith(str, 'Ti') then (
+        std.parseJson(std.rstripChars(str, 'Ti')) * std.pow(2, 40)
+      ) else (
+        str
+      )
+    );
+
+    std.split(std.toString(siToBytesDecimal(str)), '.')[0]
```

Comment:
@pracucci @jhesketh I followed your advice of fixing `siToBytes` to return an integer represented as a string), WDYT?

Reply:
If we are using `std.toString(std.parseInt(trigger.threshold))` above, do we need this here?

siToBytes could still return a number (preferably an integer, but decimals shouldn't matter if it's easier to ignore it for now). It's KEDA itself that expects a number in a string so we could do the string conversion just at that point.

Reply:
> If we are using std.toString(std.parseInt(trigger.threshold)) above, do we need this here?

@jhesketh `std.parseInt` will fail if e.g. it's a string representing a decimal, so it is necessary.

Reply:
Can't we guarantee that `siToBytesDecimal()` always return a number, then here we just use `std.ceil(siToBytesDecimal(str))`?

Reply:
@pracucci aside from the technique for converting to a number, do you prefer rounding up (`ceil`) to rounding down, since I'm currently doing the latter?

Reply:
I converted `siToBytesDecimal` to return a number, awaiting @pracucci's reply regarding `floor` vs `ceil`.